SleepEngine/Engine/Systems/System.h

```diff
@@ -0,0 +1,19 @@
+// Copyright 2019 Tihran Katolikian
+
+#pragma once
+
+BEGIN_NAMESPACE_SLEEP
+
+class Component;
+
+// System shouldn't have state, it should have a logic
+
+class System
```

Comment:
Its an interface, lets use ISystem notation for those?

Reply:
Ok